src/DataBase/EntryDatabase.cs

```diff
@@ -178,7 +181,45 @@ public async Task<IEntry> ReadAsync(Expression<Func<IEntryModel, bool>> filter,
 
         public async Task<IEntryList> ReadListAsync(int userId, EntryType entryType)
         {
-            return await ReadListAsync(x => true, userId, entryType);
+            //return await ReadListAsync(x => true, userId, entryType);
+            var list = new EntryList(entryType);
+
+            var type = entryType == 0 ? "Incomes" : "Expenses";
+
+            string query = "SELECT * FROM " + type + " WHERE UserId =" + userId;
+
+            DataSet MyDataSet = new DataSet();
+            try
+            {
+                System.Data.SqlClient.SqlDataAdapter MyDataAdapter = new System.Data.SqlClient.SqlDataAdapter(query, CnStr);
+
+                MyDataAdapter.Fill(MyDataSet);
+            }
+            catch (Exception ex)
+            {
+                return null;
+            }
+
+            DataTable table = MyDataSet.Tables[0];
+
+            List<IEntry> items = new List<IEntry>();
+            foreach (DataRow row in table.Rows)
+            { 
+                var obj = new Entry
+                {
+                    Id = row.Field<int>("Id"),
+                    Amount = row.Field<decimal>("Amount"),
+                    Date = row.Field<DateTime>("Date"),
+                    Importance = row.Field<int>("Importance"),
+                    Recurring = row.Field<bool>("IsMonthly"),
+                    Title = row.Field<string>("Title"),
+                    UserId = row.Field<int>("UserId")
+                } as IEntry;
+
+                items.Add(obj);
+            }
```

Comment:
This could be replaced with linq, and I think it would help fulfill part of the 4 requirement:

            var items = (from DataRow row in table.Rows
            select new Entry
            {
                Id = row.Field<int>("Id"),
                Amount = row.Field<decimal>("Amount"),
                Date = row.Field<DateTime>("Date"),
                Importance = row.Field<int>("Importance"),
                Recurring = row.Field<bool>("IsMonthly"),
                Title = row.Field<string>("Title"),
                UserId = row.Field<int>("UserId")
            } as IEntry).ToList();